### PR TITLE
Update Consumer Forwarding Rules in Status

### PR DESCRIPTION
**Pull request by swetharepakula**

### CODE REVIEW

**Review by freehan — COMMENTED** [review]
just nit. LGTM overall

File: pkg/psc/controller_test.go [hunk 1]
```diff
@@ -298,14 +298,81 @@ func TestServiceAttachmentCreation(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("errored adding consumer forwarding rules to gce service attachment: %q", err)
+	}
+
+	err = controller.processServiceAttachment(SvcAttachmentKeyFunc(testNamespace, saName))
+	if err != nil {
+		t.Errorf("unexpected error processing service attachment: %q", err)
+	}
+
+	updatedCR, err := controller.saClient.NetworkingV1alpha1().ServiceAttachments(testNamespace).Get(context2.TODO(), saName, metav1.GetOptions{})
```
Comment (freehan) [inline pkg/psc/controller_test.go:366]: probably worth test if changes to consudermForwardingRule will be updated

File: pkg/psc/controller.go [hunk 2]
```diff
@@ -268,15 +269,13 @@ func (c *Controller) processServiceAttachment(key string) error {
...
 			return fmt.Errorf("invalid Service Attachment Update: %q", err)
 		}
-		klog.V(4).Infof("Finished processing service attachment %s/%s", namespace, name)
-		return nil
-	}
-
-	klog.V(2).Infof("Creating service attachment %s", saName)
-	if err = c.cloud.Compute().AlphaServiceAttachments().Insert(context2.Background(), gceSAKey, gceSvcAttachment); err != nil {
-		return fmt.Errorf("failed to create GCE Service Attachment: %q", err)
+	} else {
```
[the remark this answers is not in the document]
Comment (swetharepakula) [reply pkg/psc/controller.go:272]: Insert will fail if trying to reinsert a key that already exists. I will have a followup PR that will include the Update API so the else will become more useful.

File: pkg/psc/controller_test.go (hunk 1 above)
Comment (swetharepakula) [reply pkg/psc/controller_test.go]: Added a test to ensure that the rules are update on subsequent updates

File: pkg/psc/controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (swetharepakula) [reply pkg/psc/controller.go]: For now to make it easier to read, I have just duplicated the updateServiceAttachmentStatus line and removed the else clause.

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**